### DISCUSSION TITLE
Best php file settings for big file uploads

### DISCUSSION THREAD

**Discussion by JulieVelghe** [craftcms/cms#15439] [Q&A]
Hello all,

I was wondering what the best php settings are for big file uploads in craft.
My current situation is as follows. I have website that has a connection to google cloud storage for video and audio files.
I've succesfully uploaded smaller audio files of, lets say, 8MB. However, when I try to upload a big file, 484MB, I'm getting an upload failed:
![image](https://github.com/user-attachments/assets/ec3de647-2f9f-477d-b9c6-ce5b1d5317a4)
Image above shows the message in the cms, when I try to upload via a form, I get an Internal Server Error.

My php settings seems ok to me:
```
upload_max_filesize = 512M
post_max_size = 1024M
max_input_time = 300
max_execution_time = 600
```

in general.php I have     
`->maxUploadFileSize('512M')`

Any clue what I'm missing here? I've searched the logs for any errors, but I don't find any errors that would indicate that my php values are to low.

Is there any setting in Google Cloud storage that could prevent these uploads from happening? (I don't have much experience there)

Any help is greatly appreciated.

**Comment by angrybrad** [comment]
Are you able to upload the same file to the same bucket using the same security policy using the Google Cloud CLI?

**Reply by JulieVelghe** [reply]
Hi @angrybrad , uploading the same file, with the same credentials to the same bucket via my windows powershell works perfectly fine

**Reply by angrybrad** [reply]
@JulieVelghe are those php.ini values reflected if you go to Utilities->PHP Info in the Craft control panel and search for `upload_max_filesize`, `post_max_size`, etc.?

I just did a test on a fresh Craft 5 install using a newly created GC bucket and was able to upload a 1GB file in about 2.5 minutes with no issues.

![CleanShot 2024-07-31 at 21 55 40@2x](https://github.com/user-attachments/assets/f5407ada-7ef2-4bad-a002-e088470a39e8)

php.ini changes locally were:

```
upload_max_filesize = 2G
post_max_size = 2G
max_input_time = 300
max_execution_time = 600
```

And `config/general.php` all I changed was `->maxUploadFileSize('2G')`.

**Reply by JulieVelghe** [reply]
Hi @angrybrad , I can see the adjusted values in the php-info in the craft control panel. I've also upped them to 2G, just to be sure.
Unfortunatly, It doesn't work still. The website is still Craft 4.10.2, don't know if that matters.

**Reply by JulieVelghe** [reply]
@angrybrad I've tried to upload some other files of other sizes, one of 37mb and 63mb. File of 145mb worked, my next test of 186mb failed. It seems that there is a limit somewhere between 145mb and 186mb, but I cannot find it in my php.ini.

**Reply by angrybrad** [reply]
Can you share the output of a call to `<?php phpinfo();` here or as a https://gist.github.com ? Redact out any sensitive values, or if you want to send to support@craftcms.com and reference this discussion, we can see if there are any red flags.

**Reply by JulieVelghe** [reply]
Hi @angrybrad , thanks a lot already for taking the time to help :)
I've mailed my phpinfo to support@craftcms.com
